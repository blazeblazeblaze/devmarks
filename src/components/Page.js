import React from "react";
import TopBar from "./TopBar";
import Footer from "./Footer";

import PropTypes from "prop-types";

const propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string
  ]).isRequired
};

const style = {
  backgroundColor: "#fafafa"
};

const Page = ({ children }) => (
  <div style={style}>
    <TopBar />
    {children}
    <Footer />
  </div>
);

Page.propTypes = propTypes;

export default Page;
