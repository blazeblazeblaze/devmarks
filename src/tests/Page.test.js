import React from "react";
import Page from "../components/Page";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("<Page />", () => {
  const ChildrenComponent = () => <div>children</div>;
  const tree = shallow(
    <Page>
      <ChildrenComponent />
    </Page>
  );

  it("applies background styles", () => {
    expect(tree.getElement().props.style).toEqual({
      backgroundColor: "#fafafa"
    });
  });

  it("renders given children", () => {
    expect(tree.find(ChildrenComponent).exists()).toBe(true);
  });

  it("renders Footer", () => {
    expect(tree.find("Footer").exists()).toBe(true);
  });

  it("renders given TopBar", () => {
    expect(tree.find("TopBar").exists()).toBe(true);
  });

  it("matches snapshot", () => {
    expect(toJson(tree)).toMatchSnapshot();
  });
});
