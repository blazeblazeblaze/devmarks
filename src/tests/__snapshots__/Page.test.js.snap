// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fafafa",
    }
  }
>
  <TopBar />
  <ChildrenComponent />
  <Footer />
</div>
`;
